.github/workflows/ci.yml

```diff
@@ -633,7 +634,8 @@ jobs:
       build-test-contrib-shared-tests,
       verify-aot-compat
     ]
-    if: always() && !cancelled() && !contains(needs.*.result, 'failure')
-    runs-on: windows-latest
+    if: always() && !cancelled()
+    runs-on: ubuntu-latest
     steps:
-    - run: echo 'build complete ✓'
+    - run: |
+          if ( ${{ contains(needs.*.result, 'failure') }} == true ); then echo 'build failed ✗'; exit 1; else echo 'build complete ✓'; fi
```

Comment:
@Kielek I'm kind of confused by this 😄 `build-test` is a required status. If it doesn't run (because something failed), that should block the merge by design. Having it run always seems like it would pass the status check even when there was a failure?

Reply:
Current config for `main*`:

![image](https://github.com/user-attachments/assets/cafd8851-2227-4b96-94e1-c493995b1552)

Reply:
Current status:
 When `${{ contains(needs.*.result, 'failure')` is true (one of the required step is failing) `build-status` step is not executed. It has `skipped` status. For some reasons in this case, GitHub unblocks `Merge` button. Eg. #2052 there were possibility to merge traditional way, without forcing it.

After changes:
`build-test` is always executed, but when `${{ contains(needs.*.result, 'failure')` is true it calls `exit 1`. It means that step is failing. The merge should be prevented.

Reply:
Sounds good thanks for the explanation! We should probably do same over here eh? https://github.com/open-telemetry/opentelemetry-dotnet/blob/37535a5607ee7e4056c0e274ec01d1e0111a64be/.github/workflows/ci.yml#L189

Reply:
That's the plan. Same for auto-instrumentation repo.